scrabble.js

```diff
@@ -1,8 +1,157 @@
+        "e", "e", "e", "e", "e", "e", "e", "e", "e", "e", "r", "r", "r", "r",
+        "r", "r", "f", "f", "s", "s", "s", "s", "g", "g", "g", "t", "t", "t",
+        "t", "t", "t", "h", "h", "u", "u", "u", "u", "i", "i", "i", "i", "i",
+        "i", "i", "i", "i", "v", "v", "j", "w", "w", "k", "x", "l", "l", "l",
+        "l", "y", "y", "m", "m", "z"];
+};
+
+TileBag.prototype.drawTiles = function(numOfTiles) {
+  this.randomTiles = [];
+  for(var i=0; i<numOfTiles; i++) {
+  // tiles are formatted like: [number, original index]
+    var tile = sample(this.defaultTiles);
+  // get random tiles from the bag, only push the letter to the player's tiles
+    this.randomTiles.push(tile[0]);
+  // remove those tiles from the bag so they cannot be drawn again
+  // the index 1 of tile is the random index that cause it to be drawn...
+  // this is the index in defaultTiles
+    this.defaultTiles.splice(tile[1],1);
+  }
+  return this.randomTiles;
+
+};
+
+// helper function to wrap ugly random
+function sample(array) {
+  var index = Math.floor ( Math.random() * array.length );
+  return [array[index], index];
+}
+
+Scrabble.prototype.score = function(word) {
+  this.word = word.toLowerCase(); // lowercase input for comparing
+
+    if (word === "") {
+      return 0;
+    } else {
+      var score = 0;
+      for(var i=0; i<this.word.length; i++) {
+        score += this.letterScore[this.word[i]];
+      }
+
+      if (this.word.length >= this.maxLength) {
+        score += this.bonus;
+      }
+
+    return score;
+    }
+};
+
+Scrabble.prototype.highestScore = function(wordArr) {
+  var highScore = 0;
+  var highScoreWord = "";
+
+  for(var i=0; i<wordArr.length; i++) {
+    var word = wordArr[i];
+    var score = this.score(word);
+
+    if (score > highScore) {
+      highScore = score;
+      highScoreWord = word;
+    } else if (score == highScore) {
+// if the top score is tied between multiple words,
+// pick the one with the fewest letters.
+      if (word.length < highScoreWord.length) {
+        highScoreWord = word;
+      }
+    } else {}
+  }
+
+  return highScoreWord;
+};
+
+// pass two arguments, name & game3 so that we can have multiplayer games
+var Player = function(name, game = (new Scrabble())) {
+  this.name = name;
+  this.plays = [];
+  // Each player will have their own Scrabble
+  this.scrabble = game;
+  // a new player has the maximum number of tiles available to them
+  this.tiles = this.scrabble.tileBag.drawTiles(this.scrabble.maxLength);
+};
+
+Player.prototype.play = function(word) {
+  if(this.hasWon()) {
+    return false;
+  }
+  this.word = word;
+  this.plays.push(this.word);
+
+  this.removeTiles(this.word);
+  this.drawTiles();
+
+  return this.scrabble.score(word);
+};
+
+Player.prototype.hasWon = function() {
+  return this.totalScore() > 100;
+};
+
+Player.prototype.removeTiles = function(word) {
+  this.word = word;
+  for(var i=0; i<this.word.length; i++) {
```

Comment:
Watch your spacing here, it is more pleasing to add spaces between the `=` and the `<` for the for loop